### DISCUSSION TITLE
save id SetDocument in document

### DISCUSSION THREAD

**Discussion by FrancescoIoppolo** [capawesome-team/capacitor-firebase#500] [Q&A] [answered]
Is there a method to retrieve the id of the document I'm creating to save it within the data?

const addDocument = async (image) => {
    try {
      const result =  await FirebaseFirestore.addDocument({
        reference: 'problems',
        data: { 
          **id:addDocument.id,** // example
          roomName: formData.camera, 
          problemType: formData.problemType, 
          description: formData.description,
          priority: formData.priority,
          startTimeProblem: new Date(),
          image: image,
          userId: user.nome,
          actionId: formData.problemType
        },
      });

**Comment by robingenz** [comment, marked as answer]
For example, you could first create an empty document, retrieve the ID and then update it. Or you could use the [`setDocument(...)`](https://capawesome.io/plugins/firebase/cloud-firestore/#setdocument) method and define a custom ID.

**Comment by FrancescoIoppolo** [comment]
Ok, great tips! I thought there was a way to recover it quickly. Thank you so much for the great work you do every day 💪💪💪

**Comment by FrancescoIoppolo** [comment]
If it helps, I solved it using a Firebase function.

exports.addUidproblems = functions.firestore.document('problems/{problemId}')
    .onCreate((snapshot) => {
        const problemId = snapshot.id;
        const logRef = admin.firestore().collection('problems').doc(problemId);
        return logRef.set(
            {
                problemId: problemId,
            },
            { merge: true }
        );
    });
